Modify upgrades doc (#606)
giantswarm/docs: src/content/reference/cluster-upgrades/_index.md

In particular this means:

> - Nodes are drained and recreated at the same time.

Author: ```suggestion
- Nodes are drained and recreated at the same time.
```

> Nodes will be drained, then stopped, then recreated.

Author: ```suggestion
- Nodes will be drained, then stopped, then recreated.
```

> - As a consequence of draining, Pods running on a node will be rescheduled to other nodes.

Author: Based on nodes now being drained and recreated at the same time, does this still apply?

Reviewer: I think the original sentence was fine. 

A single machine will go through this steps and it's important to mention that a machine will be drained before recreation.

Author: oh ok! put it back to the original ✌️

> - Once the master node is taken down and recreated, the Kubernetes API will be unavailable for a short time.

Author: Based on nodes now being drained and recreated at the same time, does this still apply?

Reviewer: This only applies if they are using single master

Author: can you suggest something Pau? I'm not really familiar with this area in the product :)

Author: sounds like i can leave it as is.

- As a consequence of draining, Pods running on a node will be rescheduled to other nodes.
- Once the master node is taken down and recreated, the Kubernetes API will be unavailable for a short time.

> **Note**: By default, tenant clusters have one master node each. Consider using High Availability masters setup to avoid API downtime during upgrades. (See: [High availability Kubernetes masters](https://docs.giantswarm.io/basics/ha-masters/))

Author: @paurosello WDYT?

Reviewer: looks better!

Reviewer 2: Thanks for spotting this! I missed that in my last edit.

You can even direct readers to `#checklist-ha-masters` here on the same page, not switching context too much. No need for parantheses in my opinion.

Author: @marians can you make the change please? :)

> **Note**: By default, tenant clusters have one master node each. Consider using High Availability masters setup to avoid API downtime during upgrades. (See: [High availability Kubernetes masters](https://docs.giantswarm.io/basics/ha-masters/))

Reviewer 2: ```suggestion
**Note**: By default, tenant clusters have one master node each. [Consider using a High Availability masters setup](#checklist-ha-masters) to avoid API downtime during upgrades.
```

Reviewer 2: @cokiengchiara Here you go

Author: danke!
